hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/dto/AttributeSearchRequest.kt

```diff
@@ -0,0 +1,60 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.search.services.dto
+
+import java.time.LocalDate
+
+data class AttributeSearchRequest(
+  val matchers: List<Matcher>,
+)
+
+data class Matcher(
+  val joinType: JoinType,
+  val textMatchers: List<TextMatcher>? = null,
+  val booleanMatchers: List<BooleanMatcher>? = null,
+  val integerMatchers: List<IntegerMatcher>? = null,
+  val dateMatchers: List<DateMatcher>? = null,
+  val bodyPartMatchers: List<BodyPartMatcher>? = null,
+  val children: List<Matcher>? = null,
+)
+
+data class TextMatcher(
+  val attribute: String,
+  val condition: TextCondition,
+  val searchTerm: String,
+)
+
+data class BooleanMatcher(
+  val attribute: String,
+  val condition: Boolean,
+)
+
+data class IntegerMatcher(
+  val attribute: String,
+  val minValue: Int? = null,
+  val minInclusive: Boolean = true,
```

Comment:
looking at the UI - the new figma has a few examples, but don't think we need the min or max inclusive options

Reply:
although it is difficult to tell - the latest one has an **IS** condition with cm that doesn't make any sense

Reply:
I couldn't see any examples for integer based fields. Do you think it should be LT, LTE, EQ, GT, GTE?

Reply:
difficult to tell which way they'll go. Looking at their front end for dates it has both dates on the same row, so doesn't allow inclusive (or is it exclusive) options